src/main/java/me/xmrvizzy/skyblocker/skyblock/barn/TreasureHunter.java

```diff
@@ -0,0 +1,61 @@
+package me.xmrvizzy.skyblocker.skyblock.barn;
+
+import me.xmrvizzy.skyblocker.chat.ChatFilterResult;
+import me.xmrvizzy.skyblocker.chat.ChatPatternListener;
+import me.xmrvizzy.skyblocker.config.SkyblockerConfig;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.text.Text;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+
+public class TreasureHunter extends ChatPatternListener {
+
+    private static final Map<String, String> locations;
+
+    public TreasureHunter() { super("^§e\\[NPC] Treasure Hunter§f: ([a-zA-Z, '\\-]*\\.)$"); }
+
+    @Override
+    public ChatFilterResult state() {
+        return SkyblockerConfig.get().locations.barn.solveTreasureHunter ? ChatFilterResult.FILTER : ChatFilterResult.PASS;
+    }
+
+    @Override
+    public boolean onMatch(Text message, Matcher matcher) {
+        MinecraftClient client = MinecraftClient.getInstance();
+        if (client.player == null) return false;
+        String hint = matcher.group(1);
+        String location = locations.get(hint);
+        if (location == null) return false;
+        client.player.sendMessage(Text.of("§e[NPC] Treasure Hunter§f: Go mine around " + location), false);
+        return true;
+    }
+
+    static {
+        locations = new HashMap<>();
+        locations.put("There's a treasure chest somewhere in a small cave in the gorge.", "256 70 -490");
+        locations.put("I was in the desert earlier, and I saw something near a red sand rock.", "357 82 -319");
+        locations.put("There's this guy who collects animals to experiment on, I think I saw something near his house.", "259 184 -564");
+        locations.put("There's a small house in the gorge, I saw some treasure near there.", "297 87 -562");
+        locations.put("There's this guy who says he has the best sheep in the world. I think I saw something around his hut.", "392 85 -372");
+        locations.put("I spotted something by an odd looking mushroom on one of the ledges in the Mushroom Gorge, you should check it out.", "305 73 -557");
+        locations.put("There are some small ruins out in the desert, might want to check it out.", "320 102 -471");
```

Comment:
This string should match "There are some small ruins out in the desert, might want to check them out."